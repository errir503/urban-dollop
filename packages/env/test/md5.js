/**
 * Internal dependencies
 */
const md5 = require( '../lib/md5' );

describe( 'md5', () => {
	it( 'creates a hash of a string', () => {
		const result = md5( 'hello' );
		expect( result ).toMatchSnapshot();
	} );
	it( 'creates a hash of a object', () => {
		const result = md5( { foo: 'xyz', test: { anotherFoo: 123 } } );
		expect( result ).toMatchSnapshot();
	} );
	it( 'creates a hash of a number', () => {
		const result = md5( 123789 );
		expect( result ).toMatchSnapshot();
	} );
	it( 'creates a hash of null', () => {
		const result = md5( null );
		expect( result ).toMatchSnapshot();
	} );
	it( 'uses the same hash for the same values', () => {
		const testObj = {
			test: 1,
			otherTest: {
				foo: 'hello',
			},
		};
		const result1 = md5( testObj );
		const result2 = md5( { ...testObj } );
		expect( result1 ).toBe( result2 );
	} );
} );
