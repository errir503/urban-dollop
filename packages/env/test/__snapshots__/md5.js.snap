// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`md5 creates a hash of a number 1`] = `"55587a910882016321201e6ebbc9f595"`;

exports[`md5 creates a hash of a object 1`] = `"a0294b0e88367a0ff800d78413356538"`;

exports[`md5 creates a hash of a string 1`] = `"5d41402abc4b2a76b9719d911017c592"`;

exports[`md5 creates a hash of null 1`] = `"37a6259cc0c1dae299a7866489dff0bd"`;
